### PR TITLE
Hibernate ORM no longer needs Jandex at runtime

### PR DESCRIPTION
**Pull request by Sanne**

### CODE REVIEW

File: extensions/hibernate-orm/runtime/pom.xml [hunk 1]
```diff
@@ -75,6 +75,12 @@
                     <groupId>javax.activation</groupId>
                     <artifactId>javax.activation-api</artifactId>
                 </exclusion>
+
+                <!-- Not necessary at runtime -->
+                <exclusion>
+                    <groupId>org.jboss</groupId>
+                    <artifactId>jandex</artifactId>
+                </exclusion>
```
Comment (gsmet) [inline extensions/hibernate-orm/runtime/pom.xml:83]: I'm not sure I understand this PR given its title. This was an exclusion so it was never necessary at runtime.

Do you mean that hibernate-core doesn't depend on jandex anymore?

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's clarify things before merging.

File: extensions/hibernate-orm/runtime/pom.xml (hunk 1 above)
Comment (Sanne) [reply extensions/hibernate-orm/runtime/pom.xml:83]: This is a new exclusion - I'm not removing an exclusion?

`hibernate-core` still has the dependency, as it normally makes use of it, but you won't need it in Quarkus as things are processed at deployment time

Comment (gsmet) [reply extensions/hibernate-orm/runtime/pom.xml:83]: Hu. Sorry, I don't know what happened, I somehow read it backwards. That's very odd :).

Sorry about the noise.

**Review by gsmet — APPROVED** [review]

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Is this something that can go into `2.0.x` or not?

**Comment by Sanne** [comment]
> Is this something that can go into `2.0.x` or not?

yea sounds good - not sure about how aggressive we are being with backports, I thought not since it's not a critical bugfix?   I'll label it then @gsmet can choose

**Comment by gsmet** [comment]
If it doesn't bring much, let's be conservative.

**Comment by Naros** [comment]
Hi @Sanne @gsmet does this mean if an extension implements the `AdditionalJaxbMappingProducer` interface that they'll need to explicitly include the Jandex library where-as before it was pulled in automatically?

**Comment by Sanne** [comment]
Hi @Naros ! the key is "at runtime": Hibernate ORM - when running in Quarkus - no longer need Janxed after we're done with the augmentation phase.

If another extension needs explicit access into Jandex at runtime, it should be free to add the dependency explicitly rather than relying on a transitive dependency.

But also I'd question why any extension actually would need such information at runtime; ideally and for best efficiency, the metadata should be extracted and processed during the augmentation phase.

**Comment by beikov** [comment]
@Sanne: Like pointed out on Zulip, one does not necessarily have to "use" this to "require" it due to class loading.

A Blaze-Persistence user reported an error on 2.1.x+

```
Caused by: java.lang.NoClassDefFoundError: org/jboss/jandex/IndexView
        at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
        at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
        at java.base/java.lang.Class.getMethodsRecursive(Class.java:3307)
        at java.base/java.lang.Class.getMethod0(Class.java:3293)
        at java.base/java.lang.Class.getMethod(Class.java:2106)
        at com.blazebit.persistence.integration.hibernate.Hibernate53MetadataContributor.addEntity(Hibernate53MetadataContributor.java:82)
        at com.blazebit.persistence.integration.hibernate.Hibernate53MetadataContributor.contribute(Hibernate53MetadataContributor.java:76)
        at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:292)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootMetadataBuilder.build(FastBootMetadataBuilder.java:354)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.createMetadata(PersistenceUnitsHolder.java:101)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.constructMetadataAdvance(PersistenceUnitsHolder.java:73)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.initializeJpa(PersistenceUnitsHolder.java:40)
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder$1.created(HibernateOrmRecorder.java:61)
        at io.quarkus.arc.runtime.ArcRecorder.initBeanContainer(ArcRecorder.java:70)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy_0(ArcProcessor$generateResources-1025303321.zig:128)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources-1025303321.deploy(ArcProcessor$generateResources-1025303321.zig:40)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:226)
        ... 15 more
Caused by: java.lang.ClassNotFoundException: org.jboss.jandex.IndexView
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:106)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:56)
```

**Pull request merged**